gph/bindings/ripser_bindings.cpp

```diff
@@ -15,85 +15,84 @@
 namespace py = pybind11;
 
 #if defined USE_COEFFICIENTS
-PYBIND11_MODULE(gph_ripser_coeff, m)
-{
+PYBIND11_MODULE(gph_ripser_coeff, m) {
 #else
-PYBIND11_MODULE(gph_ripser, m)
-{
+PYBIND11_MODULE(gph_ripser, m) {
 #endif
 
-    using namespace pybind11::literals;
-    m.doc() = "Ripser python interface";
+  using namespace pybind11::literals;
+  m.doc() = "Ripser python interface";
 
-    // Because `ripser` could have two different modules after compilation
-    // It's necessary to add `py::module_local()` to prevent following issue:
-    // ImportError: generic_type: type "ripserResults" is already registered!
-    // When same python module imports gtda_ripser and gtda_ripser_coeff
-    py::class_<ripserResults>(m, "ripserResults", py::module_local())
-        .def_readwrite("births_and_deaths_by_dim",
-                       &ripserResults::births_and_deaths_by_dim)
-        .def_readwrite("num_edges", &ripserResults::num_edges);
+  // Because `ripser` could have two different modules after compilation
+  // It's necessary to add `py::module_local()` to prevent following issue:
+  // ImportError: generic_type: type "ripserResults" is already registered!
+  // When same python module imports gtda_ripser and gtda_ripser_coeff
+  py::class_<ripserResults>(m, "ripserResults", py::module_local())
+      .def_readwrite("births_and_deaths_by_dim",
+                     &ripserResults::births_and_deaths_by_dim)
+      .def_readwrite("cocycles_by_dim", &ripserResults::cocycles_by_dim)
+      .def_readwrite("num_edges", &ripserResults::num_edges);
 
-    m.def(
-        "rips_dm",
-        [](py::array_t<value_t>& D, int N, int modulus, int dim_max,
-           float threshold, int num_threads) {
-            // Setup distance matrix and figure out threshold
-            auto D_ = static_cast<value_t*>(D.request().ptr);
-            std::vector<value_t> distances(D_, D_ + N);
-            compressed_lower_distance_matrix dist =
-                compressed_lower_distance_matrix(
-                    compressed_upper_distance_matrix(std::move(distances)));
+  m.def(
+      "rips_dm",
+      [](py::array_t<value_t>& D, int N, int modulus, int dim_max,
+         float threshold, int num_threads, bool do_cocycles) {
+        // Setup distance matrix and figure out threshold
+        auto D_ = static_cast<value_t*>(D.request().ptr);
+        std::vector<value_t> distances(D_, D_ + N);
+        compressed_lower_distance_matrix dist =
+            compressed_lower_distance_matrix(
+                compressed_upper_distance_matrix(std::move(distances)));
 
-            // TODO: This seems like a dummy parameter at the moment
-            float ratio = 1.0;
-            int num_edges = 0;
+        // TODO: This seems like a dummy parameter at the moment
+        float ratio = 1.0;
+        int num_edges = 0;
```

Comment:
Why not remote `ratio` if we don't use it?  We could still create an issue about it if we are not sure whether it should be implemented.

Reply:
We should remove it entirely, the operation done by `ratio` can easily be implemented in Python after the computation of PH.

Reply:
But I don't know if we do it in this PR, or we should create a PR for more misc tasks that needs to be performed ?

Reply:
I'd say this should be moved to an issue/other PR.

Reply:
Yes, we should integrate it just before releasing 0.2 IMO